core/cmd/local_client.go

```diff
@@ -127,6 +141,17 @@ func initLocalSubCmds(client *Client, devMode bool) []cli.Command {
 					Value: "/tmp/",
 				},
 			},
+			Before: func(ctx *clipkg.Context) error {
+				log.Printf("ctx %+v", *ctx)
+				s := ctx.Uint64("seconds")
+				// prevent indefinitively long profiles
+				d := time.Duration(s * uint64(time.Second))
+				max := 120 * time.Second
+				if d.Seconds() > max.Seconds() {
+					return fmt.Errorf("%w: %s is too long. profiles are limited to %s", ErrProfileTooLong, d, max)
+				}
+				return nil
+			},
```

Comment:
What is this?

Reply:
while testing i never got a timeout from the server. 

this is a safe guard to avoid super long profiles, which could negatively impact the node.

maybe there is a better way? maybe i'm over optimizing?

Reply:
:shrug: no problem, but maybe worth a separate PR?